### DISCUSSION TITLE
时间日期控件的重置问题

### DISCUSSION THREAD

**Discussion by yangzi0802** [layui/layui#1712] [问答]
### 议题条件

- [X] 我确认已查看官方使用文档：https://layui.dev ，但没有找到相关解决方案。
- [X] 我确认已在 [Issues](https://github.com/layui/layui/issues?q=is%3Aissue ) 中搜索过类似的问题，但没有找到相关解决方案。
- [X] 我已仔细阅读： 🍀[ Layui Issue 贡献指南](https://github.com/layui/layui/blob/main/CONTRIBUTING.md)

### 议题类型

功能请求

### 使用版本

v2.8.16

### 问题描述

我在使用日期控件做条件查询时，当我点击重置按钮后，日期控件直接空白，我该如何重置成刚进页面时渲染的状态？？？

### 业务代码

```auto
// 渲染日期控件
let projectYear = laydate.render({
            elem: '#projectYear',
            type: 'year',
            value: new Date().getFullYear(),
            format: 'yyyy'
        })
// 重置
form.on('submit(reset)', function (data) {
        // ？？？？？
        });
```

### 截图补充

刚进页面时
![image](https://github.com/layui/layui/assets/28970398/12931c17-12ea-4a3d-96f3-bcca27cfbc9b)
点击重置后
![image](https://github.com/layui/layui/assets/28970398/ce95ccc5-ffca-4528-82f3-2679b8afdfe1)

### 浏览器

chrome

### 友好承诺

- [X] 我承诺将本着相互尊重、理解和友善的态度进行交流，共同维护 Layui 良好的社区氛围。

**Comment by Sight-wcg** [comment]
```js
<input type="text" class="layui-input" id="ID-laydate-more-value" value="2014">

  laydate.render({
    elem: '#ID-laydate-more-value',
    type: 'year',
    value: 2014,
    isInitValue: true,
    format: 'yyyy'
  });
```

**Reply by yangzi0802** [reply]
必须要在input value写默认值？ 我想根据当前年份 或者后台获取的年份当默认值 就不行了吧？

**Reply by Sight-wcg** [reply]
```  
laydate.render({
    elem: '#el',
    value: new Date().getFullYear(),
  });
  $('#el').attr('value', new Date().getFullYear())
```

**Reply by yangzi0802** [reply]
大佬 还有一个问题 
这段代码是表格的before事件
before(options) {
                let condition = {};
                condition.projectYear = projectYear.config.dateTime?.year ?? projectYear.config.value;
                if (tabId == 1) {
                    condition.projectStatus = 'status.20';
                }
                options.where = condition;
            }
按你上面的手动attr 但是重置后 laydate.config.dateTime没更新 我需要手动更新？
![Uploading image.png…]()

**Reply by yangzi0802** [reply]
//监听重置
form.on('submit(reset)', function (data) {
    projectYear.config.dateTime.year = projectYear.config.value;
    form.render('select')
    table.reload('awardInfoTable');
});
好像只能这样改了

**Comment by yangzi0802** [comment]
首先第一次table加载时，并没有条件查询 layuidate对象没有 config.dateTime属性 只能取config.value值
当重置后layuidate控件直接变成空白 只能手动在input添加value属性 或者js 添加
let projectYear = laydate.render({
            elem: '#projectYear',
            type: 'year',
            value: new Date().getFullYear(),
            isInitValue: true,
            format: 'yyyy'
        })
// 需要添加这行 或后台取值
$('#projectYear').attr('value', new Date().getFullYear())
但是重置后 并没有修改config.dateTime的值，只能在重置方法中手动修改
//监听重置
  form.on('submit(reset)', function (data) {
      projectYear.config.dateTime.year = projectYear.config.value;
      table.reload('awardInfoTable');
  });
这样是可以用的
